sdk/docs/iot/provisioning.md

```diff
@@ -8,7 +8,7 @@ For more details about Provisioning Service check [Azure documentation](https://
 
 - Please refer to the state machine diagram which describes the Azure SDK for Embedded C flow for Provisioning Service [here](./mqtt_state_machine.md#device-provisioning-service).
 
-- **Samples**: [This page](https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/samples/iot/provisioning/) explains samples for the Azure SDK for Embedded C IoT Provisioning Client and how to use them.
+- **Samples**: [This page](https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/README.md) explains samples for the Azure SDK for Embedded C IoT Provisioning Client and how to use them.
```

Comment:
Double check this is the link we want to route to @danewalton, @momuno for provisioning. I know we had recent changes in this space, around samples.

Reply:
https://github.com/Azure/azure-sdk-for-c/tree/master/sdk/samples/iot/    will direct you to the page with the readme.  

We should be consistent if we are linking to the readme directly or to the page that will source the readme.  
Example:  https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/  (further above) vs. https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/samples/iot/README.md  (this change)

Reply:
I would also update the explanations to match:
"explains samples for the Azure SDK for Embedded C and how to use them." vs "explains samples for the Azure SDK for Embedded C IoT Provisioning Client and how to use them."

The first quotation should suffice for both.

Reply:
That sounds great. Please update the wording to your liking in a separate PR, @momuno when you get the chance,

Reply:
The link change in this PR matches more closely to the wording atm (page vs directory), so I think the PR is OK the way it is, for now. There is also something nice about the navigation/readability if we point to a specific targeted location so the user knows what to read next, rather than a directory with a bunch of files in it (granted one of the file being a README, especially in GitHub, makes that easier, since the user just scrolls down to see the content).